toutv/bos.py

```diff
@@ -58,9 +69,19 @@ def get_proxies(self):
 
     def _do_request(self, url, timeout=None, params=None):
         proxies = self.get_proxies()
+        auth = self.get_auth()
 
         try:
-            r = requests.get(url, params=params, headers=toutv.config.HEADERS,
+            headers = dict(toutv.config.HEADERS)
+
+            if auth and params:
+                url = toutv.config.TOUTV_AUTH_PLAYLIST_URL
```

Comment:
I don't really like the fact that we overwrite the url passed to the function. Could it be cleaner?

What are the conditions that define which url a request should use?